pkg/controller/network/expressroutecircuits/managed.go

```diff
@@ -0,0 +1,161 @@
+}
+
+type connecter struct {
+	client client.Client
+}
+
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := azurenetwork.NewExpressRouteCircuitsClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{client: cl}, nil
+}
+
+type external struct {
+	client networkapi.ExpressRouteCircuitsClientAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotExpressRouteCircuits)
+	}
+
+	az, err := e.client.Get(ctx, exp.Spec.ResourceGroupName, meta.GetExternalName(exp))
+
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetExpressRouteCircuits)
+	}
+
+	network.UpdateExpressRouteCircuitStatusFromAzure(exp, az)
+	exp.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotExpressRouteCircuits)
+	}
+
+	exp.Status.SetConditions(xpv1.Creating())
+
+	ercParams := network.NewExpressRouteCircuitsParameters(exp)
+	if _, err := e.client.CreateOrUpdate(ctx, exp.Spec.ResourceGroupName, meta.GetExternalName(exp), ercParams); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateExpressRouteCircuits)
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotExpressRouteCircuits)
+	}
+
+	az, err := e.client.Get(ctx, exp.Spec.ResourceGroupName, meta.GetExternalName(exp))
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errGetExpressRouteCircuits)
+	}
+
+	if network.ExpressRouteCircuitNeedsUpdate(exp, az) {
+		ercParams := network.NewExpressRouteCircuitsParameters(exp)
+		if _, err := e.client.CreateOrUpdate(ctx, exp.Spec.ResourceGroupName, meta.GetExternalName(exp), ercParams); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateExpressRouteCircuits)
+		}
+	}
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+	exp, ok := mg.(*v1alpha3.ExpressRouteCircuits)
+	if !ok {
+		return errors.New(errNotExpressRouteCircuits)
+	}
+
+	mg.SetConditions(xpv1.Deleting())
```

Comment:
```suggestion
```
crossplane-runtime can now [handle setting "creating" and "deleting" conditions](https://github.com/crossplane/crossplane-runtime/pull/242).